### PR TITLE
WDK installation fix for win19

### PR DESCRIPTION
**Pull request by miketimofeev**
We have several issues related to WDK:
https://github.com/actions/virtual-environments/issues/259
https://github.com/actions/virtual-environments/issues/428
The root cause is an incorrect installation order - WDK is supposed to be installed after VS19, not before so that the extension can be added to VS.

Changes:
- Installation order changed
- Try-catch on package installation added in order to break the build if the package is not installed
- Validate-WDK script works properly now, previously it's just put hardcoded value to the softwarelist(the same can be done for windows16 if needed)
- Get-VS19ExtensionVersion function created to use it in both Validate-WDK and Validate-Wix scripts

### CODE REVIEW

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 1]
```diff
@@ -160,3 +160,31 @@ Hashtable for service arguments
...
+function Get-VS19ExtensionVersion
+{
+    param (
+        [string] [Parameter(Mandatory=$true)] $packageName
+    )
+
+    $vsProgramData = Get-Item -Path "C:\ProgramData\Microsoft\VisualStudio\Packages\_Instances"
+    $instanceFolders = Get-ChildItem -Path $vsProgramData.FullName
+
+    if($instanceFolders -is [array])
```
Comment (vmapetr) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1]: nitpick: missing space `if ($instanceFolders -is [array])`

Comment (miketimofeev) [reply images/win/scripts/ImageHelpers/InstallHelpers.ps1]: thanks! Fixed

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 2]
```diff
@@ -160,3 +160,31 @@ Hashtable for service arguments
...
     }
 }
+
+function Get-VS19ExtensionVersion
+{
+    param (
+        [string] [Parameter(Mandatory=$true)] $packageName
+    )
+
+    $vsProgramData = Get-Item -Path "C:\ProgramData\Microsoft\VisualStudio\Packages\_Instances"
```
Comment (al-cheb) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1]: What is the purpose Get-Item -Path ? We already have fullName.

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 3]
```diff
@@ -160,3 +160,31 @@ Hashtable for service arguments
...
+    $instanceFolders = Get-ChildItem -Path $vsProgramData.FullName
```
Comment (al-cheb) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1]: .FullName is not necessary

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 [hunk 4]
```diff
@@ -160,3 +160,31 @@ Hashtable for service arguments
...
+
+    if ($instanceFolders -is [array])
+    {
+        Write-Host "More than one instance installed"
+        exit 1
+    }
+
+    $stateContent = Get-Content -Path ($instanceFolders.FullName + '\state.packages.json')
```
Comment (al-cheb) [inline images/win/scripts/ImageHelpers/InstallHelpers.ps1]: Use Join-Path

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 (hunk 3 above)
Comment (miketimofeev) [reply images/win/scripts/ImageHelpers/InstallHelpers.ps1]: Thanks, removed

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 (hunk 2 above)
Comment (miketimofeev) [reply images/win/scripts/ImageHelpers/InstallHelpers.ps1]: Thanks, removed

File: images/win/scripts/ImageHelpers/InstallHelpers.ps1 (hunk 4 above)
Comment (miketimofeev) [reply images/win/scripts/ImageHelpers/InstallHelpers.ps1]: Switched, thanks

**Pull request merged**